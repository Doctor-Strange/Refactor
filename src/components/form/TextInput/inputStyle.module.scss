@import "../../../styles/colors";

.text_input_label {
  color: $Main_black_Text_Color;
  text-align: right;
  font-family: inherit;
  font-size: 1.4rem ;
}
.text_input {
  width: 100%;
  display: block;
  font-family: inherit;
  margin: 0;
  outline: 0;
  -webkit-appearance: none;
  line-height: 1.5;
  padding: 1rem;
  font-size: 1.4em;
  background: #fff;
  border: 1px solid rgba(34, 36, 38, 0.15);
  color: rgba(0, 0, 0, 0.87);
  border-radius: 0.4rem;
  box-shadow: 0 0 0 0 transparent inset;
}
