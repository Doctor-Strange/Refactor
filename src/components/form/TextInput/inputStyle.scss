@import "../../../styles/colors";

.text_input_container {
  width: 100%;
  display: block;
  position: relative;
  .text_input_label {
    color: $Main_black_Text_Color;
    text-align: right;
    font-family: inherit;
    font-size: 1.4rem;
  }
  .text_input {
    font-family: inherit;
    margin: 1rem auto;
    display: block;
    width: 100%;
    outline: 0;
    -webkit-appearance: none;
    line-height: 1.5;
    padding: 1rem;
    font-size: 1.4em;
    background: #fff;
    border: 1px solid $input_border;
    color: rgba(0, 0, 0, 0.87);
    border-radius: 0.4rem;
    box-shadow: 0 0 0 0 transparent inset;
    &:invalid {
      border: 1px solid #d83030 !important;
      background: rgba(216, 48, 48, 0.1) !important;
    }
  }
  .clean_icon {
    position: absolute;
    left: 10px;
    top: 36px;
  }
  .input_error_message {
    font-size: 1.2rem;
    color: $Error_text_color;
  }
}
