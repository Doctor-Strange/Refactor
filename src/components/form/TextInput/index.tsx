import React, { useRef, useEffect } from "react";
import "./inputStyle.scss";
import { IoMdClose } from "react-icons/io";

const TextInput = (props: ItextInput) => {
  const TextInput = useRef(null);

  const ValueHandler = (e) => {
    let value = e.target.value;
    if (props.number) {
      const persianNumbers = [
        /۰/g,
        /۱/g,
        /۲/g,
        /۳/g,
        /۴/g,
        /۵/g,
        /۶/g,
        /۷/g,
        /۸/g,
        /۹/g,
      ];
      const arabicNumbers = [
        /٠/g,
        /١/g,
        /٢/g,
        /٣/g,
        /٤/g,
        /٥/g,
        /٦/g,
        /٧/g,
        /٨/g,
        /٩/g,
      ];
      for (var i = 0; i < 10; i++) {
        value = value
          .replace(persianNumbers[i], i)
          .replace(arabicNumbers[i], i);
      }
      value = value.replace(/[^0-9]/g, "");
      props.onChangeHandler(value);
    } else props.onChangeHandler(value);
  };

  useEffect(() => {
    if (props.error.status) {
      scrollTo(0, TextInput.current.offsetTop);
    }
  }, [props.error]);

  return (
    <div className="text_input_container" ref={TextInput}>
      <label
        style={{
          color: props.LabelColor,
        }}
        className="text_input_label"
      >
        {props.label}
      </label>
      <input
        onInvalid={(e: any) => {
          if (props.value.length < props.min) {
            e.target.setCustomValidity(
              `حداقل ورودی باید ${props.min} کاراکتر باشد`
            );
          } else if (props.value.length > props.max) {
            e.target.setCustomValidity(
              `طول ورودی نباید بیشتر از ${props.max} کاراکتر باشد`
            );
          }
        }}
        autoFocus={props.autoFocus}
        className={[
          "text_input",
          props.error.status ? "inputError" : null,
        ].join(" ")}
        name={props.name}
        value={
          props.number
            ? props.value === ""
              ? props.value.toLocaleString()
              : props.localeString
              ? props.value
              : Number(props.value).toLocaleString()
            : props.value
          // props.value
        }
        onChange={(e: any) => {
          e.target.setCustomValidity("");
          ValueHandler(e);
        }}
        disabled={props.disabled}
        maxLength={props.max}
        minLength={props.min}
        placeholder={props.placeholder}
      />
      {props.value.length > 0 && !props.HideClearIcon && (
        <IoMdClose
          color="rgb(165, 165, 165)"
          size="2rem"
          className="clean_icon"
          onClick={() => props.clearField()}
        />
      )}
      {props.error.status && (
        <p className="input_error_message">{props.error.message}</p>
      )}
    </div>
  );
};

interface ItextInput {
  name: string;
  clearField: any;
  onChangeHandler: any;
  value: string;
  error: any;
  autoFocus: boolean;
  disabled?: boolean;
  max?: number;
  min?: number;
  label?: string;
  placeholder?: string;
  LabelColor?: string;
  number?: boolean;
  HideClearIcon?: boolean;
  localeString?: boolean;
}

export default TextInput;
