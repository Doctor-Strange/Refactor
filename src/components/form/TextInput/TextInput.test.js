import React from "react";
import { mount } from "enzyme";
import TextInput from "./";

const props = {
  name: "input",
  clearField: () => {},
  onChangeHandler: () => {},
  value: "",
  error: { status: false, message: "" },
  autoFocus: false,
};
const wrapper = mount(<TextInput {...props} />);

describe("Text Input component", () => {
  it("Text Input snap shot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
