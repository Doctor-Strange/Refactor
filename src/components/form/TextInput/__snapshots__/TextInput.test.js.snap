// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Input component Text Input snap shot 1`] = `ReactWrapper {}`;
